### PR TITLE
Remove unused float serialization

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

**Comment by laanwj** [comment]
Good catch!

Code review ACK fa57a8fed196c5e89659265e1b39efb0e2bec9d0

**Comment by practicalswift** [comment]
Nice!

cr ACK fa57a8fed196c5e89659265e1b39efb0e2bec9d0

**Comment by sipa** [comment]
Concept ACK, certainly no need to keep this longer if it's unused.

File: src/serialize.h [hunk 1]
```diff
@@ -252,7 +239,7 @@ template<typename Stream> inline void Unserialize(Stream& s, int32_t& a ) { a =
 template<typename Stream> inline void Unserialize(Stream& s, uint32_t& a) { a = ser_readdata32(s); }
 template<typename Stream> inline void Unserialize(Stream& s, int64_t& a ) { a = ser_readdata64(s); }
 template<typename Stream> inline void Unserialize(Stream& s, uint64_t& a) { a = ser_readdata64(s); }
-template<typename Stream> inline void Unserialize(Stream& s, float& a   ) { a = ser_uint32_to_float(ser_readdata32(s)); }
+template<typename Stream> inline void Unserialize(Stream& s, float& a   ) { static_assert(ALWAYS_FALSE<Stream>, "Not implemented"); }
```
[the remark this answers is not in the document]
Comment (laanwj) [reply src/serialize.h:242]: I kind of hope we can remove the double case as well some time soon. But I think this is fine for now.

[the remark this answers is not in the document]
Comment (sipa) [reply src/serialize.h:242]: I'm fine with doing this or not. Even if it's accidentally invoked, at worst it'll result in serialization as a double, which is just 4 bytes bigger, but no loss compared to (highly hypothetical) expected float serialization.

[the remark this answers is not in the document]
Comment (MarcoFalke) [reply src/serialize.h:242]: Should be easy to remove it in a follow-up. E.g https://github.com/bitcoin/bitcoin/pull/21966#issuecomment-842528708

**Comment by laanwj** [comment]
I think this can be closed as it is included in #21966.

**Comment by laanwj** [comment]
Closing as #21966 was merged.

**Pull request closed without merging**